source/docs/software/commandbased/commands-v3/what-is-commands-v3.rst

```diff
@@ -0,0 +1,132 @@
+# What is Commands v3?
+
+.. todo:: This article introduces the conceptual foundation of Commands v3, explaining what problem it solves and its core concepts.
+
+## Introduction
+
+.. todo::
+   - Recap the command-based pattern (commands + mechanisms for resource management)
+   - Explain what's fundamentally different in v3 vs v2
+   - Why imperative style was chosen over declarative composition
+   - Brief history: evolution from v1 → v2 → v3
+
+## The Problem v3 Solves
+
+.. todo::
+   **Learning Curve Issues:**
+   - v2 requires understanding functional composition, decorators, and lambda chaining
+   - New programmers struggle with splitting logic across initialize/execute/isFinished
+   - Common mistake: writing blocking loops in execute() that stall the scheduler
+
+   **Code Readability:**
+   - Complex behaviors require deeply nested decorator chains
+   - Hard to see the sequential flow of actions
+   - Difficult to add conditional logic mid-sequence
+
+   **Example comparison:** Show a complex autonomous routine in v2 (decorator chains) vs v3 (sequential imperative code) to illustrate the readability difference.
+
+## Core Concepts
+
+### Commands as Imperative Functions
+
+.. todo::
+   - Commands in v3 are written as normal sequential functions
+   - Use loops, if-statements, and control flow naturally
+   - The ``Coroutine`` object provides pause points via ``yield()``
+   - Example: Simple command that reads like pseudocode
+
+### Coroutines and Cooperative Multitasking
+
+.. todo::
+   - What are coroutines (functions that can pause and resume)
+   - How Java's ``Continuation`` API enables this (internal implementation detail)
+   - Cooperative vs preemptive multitasking
+   - Commands must voluntarily yield control
+   - The scheduler cannot forcibly suspend commands
+
+### Mechanisms as Exclusive Resources
+
+.. todo::
+   - Mechanisms represent hardware groupings (like v2 Subsystems)
+   - Only one command can require a mechanism at a time
+   - Automatic resource conflict detection and resolution
+   - Comparison to v2 Subsystems with key differences
+
+### Command Priorities
+
+.. todo::
+   - Priority levels replace v2's binary interrupt behavior
```

Comment:
Just documenting priority levels is fine, comparisons can be in the migration guide